backend/stakepoold/rpc/rpcserver/client.go

```diff
@@ -0,0 +1,168 @@
+// disconnected and the channel is not yet blocking, creates a new channel that
+// will be closed on a successful reconnect.
+func (c *Client) Connected() <-chan struct{} {
+	c.connectedMux.Lock()
+	defer c.connectedMux.Unlock()
+	if c.RPCClient().Disconnected() {
+		// Start blocking on connected chan if not already.
+		select {
+		case <-c.connected:
+			c.connected = make(chan struct{})
+		default:
+		}
+	}
+	return c.connected
+}
+
+// RPCClient allows access to the underlying rpcclient by providing a copy of
+// its address.
+func (c *Client) RPCClient() *rpcclient.Client {
+	c.mux.RLock()
+	defer c.mux.RUnlock()
+	return c.client
+}
+
+// New creates a new Client and starts the automatic reconnection handler.
+// Returns an error if unable to construct a new rpcclient.
+func New(cfg *rpcclient.ConnConfig, ntfnHandlers *rpcclient.NotificationHandlers) (*Client, error) {
+	client, err := rpcclient.New(cfg, ntfnHandlers)
+	if err != nil {
+		return nil, err
+	}
+	c := &Client{
+		client:       client,
+		cfg:          cfg,
+		ntfnHandlers: ntfnHandlers,
+		stop:         make(chan struct{}),
+		connected:    make(chan struct{}),
+	}
+	// A closed connected channel indcates successfully connected.
+	close(c.connected)
+	go func(c *Client) {
+		c.wg.Add(1)
+		c.autoReconnect()
+	}(c)
+	return c, nil
+}
+
+// Stop stops automatic reconnections.
+func (c *Client) Stop() {
+	c.stop <- struct{}{}
+	c.wg.Wait()
+	// Stop blocking on connected if blocking, as we will never reconnect again.
+	select {
+	case <-c.connected:
+	default:
+		close(c.connected)
+	}
+}
+
+// autoReconnect waits for a disconnect or stop. On disconnect it attempts to
+// reconnect to the client every connectionRetryInterval.
+//
+// This function must be run as a goroutine.
+func (c *Client) autoReconnect() {
+out:
+	for {
+		select {
+		case <-c.stop:
+			break out
+		case <-time.After(disconnectCheckInterval):
+			if !c.RPCClient().Disconnected() {
+				continue out
+			}
+			// Client is disconnected. Try to reconnect.
+			// Start blocking on connected chan if not already.
+			c.connectedMux.Lock()
+			select {
+			case <-c.connected:
+				c.connected = make(chan struct{})
+			default:
+			}
+			c.connectedMux.Unlock()
+		}
+
+	reconnect:
+		for {
+			select {
+			case <-c.stop:
+				break out
+			default:
+			}
+
+			// Start a new client.
+			client, err := rpcclient.New(c.cfg, c.ntfnHandlers)
+
+			if err != nil {
+				log.Warnf("Failed to connect to %s: %v",
+					c.cfg.Host, err)
+
+				log.Infof("Retrying connection to %s in "+
+					"%s", c.cfg.Host, connectionRetryInterval)
+				time.Sleep(connectionRetryInterval)
+				continue reconnect
+			}
+
+			// Properly shutdown old client.
+			c.mux.Lock()
+			c.client.Shutdown()
+			c.client.WaitForShutdown()
+			// Switch the new client with the old, shutdown one.
+			c.client = client
+			c.mux.Unlock()
+
+			// Close the connected channel so that all waiting
+			// processes can continue.
```

Comment:
80 char wrap of all comments for consistency

Reply:
Isn't it?

![80](https://user-images.githubusercontent.com/34998433/64745375-9ff98b00-d542-11e9-9545-26afd2490145.PNG)